packages/picasso/src/RichTextEditor/RichTextEditor.tsx

```diff
@@ -0,0 +1,237 @@
+  /**
+   * This Boolean attribute indicates that the user cannot interact with the control.
+   */
+  disabled?: boolean
+  /** unique identifier */
+  id: string
+  /**
+   * Indicate wether the editor is in an error state
+   */
+  error?: boolean
+  /**
+   * The maximum number of characters that the user can enter.
+   * If this value isn't specified, the user can enter an unlimited
+   * number of characters.
+   */
+  maxLength?: number
+  /**
+   * The minimum number of characters required that the user should enter.
+   */
+  minLength?: number
+  /**
+   * Custom counter message for minLength
+   */
+  minLengthMessage?: CounterMessageSetter
+  /**
+   * Custom counter message for maxLength
+   */
+  maxLengthMessage?: CounterMessageSetter
+  /**
+   * Callback on text change
+   */
+  onChange?: (value: string) => void
+  /**
+   * Callback for blur event
+   */
+  onBlur?: () => void
+  /**
+   * Callback for focus event
+   */
+  onFocus?: () => void
+  /** The placeholder attribute specifies a short hint that describes the expected value of a text editor. */
+  placeholder?: string
+  testIds?: {
+    wrapper?: string
+    editor?: string
+    headerSelect?: string
+    boldButton?: string
+    italicButton?: string
+    unorderedListButton?: string
+    orderedListButton?: string
+  }
+}
+
+const useStyles = makeStyles<Theme>(styles, {
+  name: 'RichTextEditor'
+})
+
+export const RichTextEditor = forwardRef<HTMLDivElement, Props>(
+  function RichTextEditor(
+    {
+      'data-testid': dataTestId,
+      autoFocus = false,
+      className,
+      defaultValue,
+      disabled,
+      id,
+      onChange = noop,
+      onFocus = noop,
+      onBlur = noop,
+      placeholder,
+      minLength,
+      maxLength,
+      minLengthMessage,
+      maxLengthMessage,
+      style,
+      error,
+      testIds
+    },
+    ref
+  ) {
+    const classes = useStyles()
+    const toolbarRef = useRef<HTMLDivElement>(null)
+    const editorRef = useRef<HTMLDivElement>(null)
+    const wrapperRef = useRef<HTMLDivElement>(null)
+    const { dispatch, state } = useTextEditorState()
+
+    const { handleSelectionChange } = useOnSelectionChange({ dispatch })
+    const { handleTextFormat } = useOnTextFormat({ dispatch })
+    const {
+      handleBold,
+      handleItalic,
+      handleHeader,
+      handleOrdered,
+      handleUnordered
+    } = useToolbarHandlers({
+      editorRef,
+      handleTextFormat,
+      format: state.toolbar.format
+    })
+
+    const { isEditorFocused, handleFocus, handleBlur } = useOnFocus({
+      autoFocus,
+      editorRef,
+      toolbarRef,
+      wrapperRef,
+      onFocus,
+      onBlur,
+      dispatch
+    })
+
+    const defaultValueInHtml = useMemo(
+      // this effects needs to happen only once on first render
+      () =>
+        defaultValue
+          ? hastUtilToHtml(hastSanitize(defaultValue))
+          : defaultValue,
+      /* eslint-disable-next-line */
+      []
+    )
```

Comment:
Instead of using eslint ignore comments we could refactor this to use an `useState` hook, it accepts a function as first argument that is only run when the hook is created. We can ignore the setter

```suggestion
    const [defaultValueInHtml] = useState(
      // this effects needs to happen only once on first render
      () =>
        defaultValue
          ? hastUtilToHtml(hastSanitize(defaultValue))
          : defaultValue
    )
```

Reply:
Agree, only input value should not be function in this case 👍